cmd/clusterctl/client/cluster/cert_manager.go

```diff
@@ -236,12 +236,28 @@ func (cm *certManagerClient) PlanUpgrade(ctx context.Context) (CertManagerUpgrad
 	}, nil
 }
 
+// getFilteredCertManagerResources gets all relevant objects for a cert-manager deployment
+// it also includes objects in the kube-system namespace because leader election namespace
+// is hard coded to kube-system https://github.com/cert-manager/cert-manager/issues/6716
+// it also deletes components that are created by Kubernetes and won't be used to see if
+// new objects need to be created for a cert-manager upgrade.
+func (cm *certManagerClient) getFilteredCertManagerResources(ctx context.Context) ([]unstructured.Unstructured, error) {
+	objs, err := cm.proxy.ListResources(ctx, map[string]string{clusterctlv1.ClusterctlCoreLabel: clusterctlv1.ClusterctlCoreLabelCertManagerValue}, certManagerNamespace, metav1.NamespaceSystem)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed get cert manager components")
+	}
+	objs = slices.DeleteFunc(objs, func(obj unstructured.Unstructured) bool {
```

Comment:
the reason we are deleting this is because when we include these created objects by the API this condition below always be true because they are created by kubernetes API.

https://github.com/kubernetes-sigs/cluster-api/blob/f8a2970e9c8516f2c3d5cad4985543314a215a6b/cmd/clusterctl/client/cluster/cert_manager.go#L377